### PR TITLE
[Puppeteer] Support acceptPopup, cancelPopup, seeInPopup and grabPopupText functions

### PR DESCRIPTION
**Pull request by reubenmiller**
#### Goal
Expand the Puppeteer Helper to support the popup functions to more towards feature parity with the WebDriverIO Helper so the two can be used interchangeably.

The full proposal and design decisions can be found in #874

#### Changes
[Puppeteer] Adding support for I.acceptPopup, I.cancelPopup, I.seeInPopup, I.grabPopupText and two Puppeteer specific functions: I.amAcceptingPopups and I.amCancellingPopups
[WebDriverIO] Adding `I.grabPopupText()`

### CODE REVIEW

File: lib/helper/Puppeteer.js [hunk 1]
```diff
@@ -95,6 +165,114 @@ class Puppeteer extends Helper {
...
+   * ```
+   */
+  amAcceptingPopups() {
+    popupStore.actionType = 'accept';
+  }
+
+  /**
+   * Accepts the active JavaScript native popup window, as created by window.alert|window.confirm|window.prompt.
+   * Don't confuse popups with modal windows, as created by [various
+   * libraries](http://jster.net/category/windows-modals-popups). Appium: support only web testing
```
Comment (DavertMik) [inline lib/helper/Puppeteer.js]: Please remove mentioning of Appium

File: lib/helper/Puppeteer.js [hunk 2]
```diff
@@ -95,6 +165,114 @@ class Puppeteer extends Helper {
...
+   * I.cancelPopup();
+   * ```
+   */
+  amCancellingPopups() {
+    popupStore.actionType = 'cancel';
+  }
+
+  /**
+   * Dismisses the active JavaScript popup, as created by window.alert|window.confirm|window.prompt.
+   * Appium: support only web testing
```
Comment (DavertMik) [inline lib/helper/Puppeteer.js]: There is no Appium here :)

File: lib/helper/Puppeteer.js [hunk 3]
```diff
@@ -14,6 +14,74 @@ const ElementNotFound = require('./errors/ElementNotFound');
 
 const puppeteer = requireg('puppeteer');
 
+/**
+ * Class to handle the interaction with the Dialog (Popup) Class from Puppeteer
+ */
+class PopupStore {
```
Comment (DavertMik) [inline lib/helper/Puppeteer.js]: Please move this to a separate file. Maybe to `lib/helper/extra/Popup.js` or smth like that

**Comment by reubenmiller** [comment]
Thanks for the suggestions, I've implemented all of them :)

**Comment by DavertMik** [comment]
Thank you! Awesome work. Popups are coming!
I'm going to release the next version soon

**Pull request merged**